dotcom-rendering/src/lib/ad-targeting.ts

```diff
@@ -10,9 +12,10 @@ export const buildAdTargeting = (
 		};
 	}
 	const { config } = CAPI;
+	const isSignedIn = (typeof window !== "undefined") && getCookie('GU_U') ? 't' : 'f';
```

Comment:
Will this work on the server? I think `getCookie` reads `document`

Reply:
Also, as far as I can tell, this adds a blocking synchronous cookie read to `App.tsx`. Does this then slow hydration for everything?

Reply:
We already read this cookie in `App.tsx` but inside a useEffect, like this:

```	useEffect(() => {
		setIsSignedIn(!!getCookie('GU_U'));
	}, []);```
	
Is there perhaps a way to use that state value asynchronously instead?